docs/configure/outputs/logstash.asciidoc

```diff
@@ -11,209 +11,181 @@ image:./binary-yes-fm-no.svg[supported deployment methods]
 
 output {
-    elasticsearch {
-        hosts => ["http://localhost:9200"]
-        index => "%{[@metadata][index]}"
-    }
-}
-------
-<1> Creates a new field named `@metadata.index`.
-`%{[@metadata][beat]}` sets the first part of the index name to the value of the `metadata.beat` field.
-`%{[@metadata][version]}` sets the second part to {beatname_uc}'s version.
-`%{[processor][event]}` sets the final part based on the APM event type.
-For example: +{beat_default_index_prefix}-{version}-sourcemap+.
-<2> In addition to the above rules, this pattern appends a date to the `index` name so {ls} creates a new index each day.
-For example: +{beat_default_index_prefix}-{version}-transaction-{sample_date_0}+.
-endif::[]
-
-Events indexed into {es} with the {ls} configuration shown here
-will be similar to events directly indexed by {beatname_uc} into {es}.
-
-ifndef::apm-server[]
-NOTE: If {ilm-init} is not being used, set `index` to `%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}` instead so {ls} creates an index per day, based on the `@timestamp` value of the events coming from {beats}.
-endif::[]
-
-ifdef::apm-server[]
-=== {ls} and {ilm-init}
-
-When used with {apm-guide-ref}/ilm-how-to.html[{ilm-cap}], {ls} does not need to create a new index each day.
-Here's a sample {ls} configuration file that would accomplish this:
-
-[source,logstash]
-------
-input {
-    beats {
-        port => 5044
-    }
+  elasticsearch {
+    data_stream => "true"
+    cloud_id => "YOUR_CLOUD_ID_HERE"
+    cloud_auth => "YOUR_CLOUD_AUTH_HERE"
+  }
 }
+----
 
-output {
-    elasticsearch {
-        hosts => ["http://localhost:9200"]
-        index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{[processor][event]}" <1>
+[float]
+== Accessing the @metadata field
+
+Every event sent to {ls} contains a special field called
+{logstash-ref}/event-dependent-configuration.html#metadata[`@metadata`] that you can
+use in {ls} for conditionals, filtering, indexing and more.
+APM Server sends the following @metadata to {ls}:
+
+["source","json",subs="attributes"]
+----
+{
+    ...
+    "@metadata": {
+      "beat": "apm-server", <1>
+      "version": "{version}" <2>
     }
 }
-------
-<1> Outputs documents to an index:
-`%{[@metadata][beat]}` sets the first part of the index name to the value of the `metadata.beat` field.
-`%{[@metadata][version]}` sets the second part to {beatname_uc}'s version.
-`%{[processor][event]}` sets the final part based on the APM event type.
-For example: +{beat_default_index_prefix}-{version}-sourcemap+.
-endif::[]
+----
+<1> To change the default `apm-server` value, set the
+<<logstash-index,`index`>> option in the APM Server config file.
+<2> The current version of APM Server.
+
+In addition to @metadata, APM Server provides other potentially useful fields, like the
+`processor.event` field and `data_stream` fields, which can be used to conditionally operate on
+{apm-guide-ref}/data-model.html[event types], namespaces, or datasets.
```

Comment:
Reviewers: I removed the custom index example because it was outdated. It would be great to come up with a new example to provide here. What is a common use-case you see with Logstash?

Reply:
I believe using logstash for routing events to different data streams can still be a fair use case; e.g. routing all `metrics` events to the same metrics datastream, rather than to service specific ones for custom metrics.

Reply:
Thanks! I was able to get this working with the following config. I've updated the documentation to match.

```
output {
  if [@metadata][beat] == "apm-server" {
    if [processor][event] == "metric" {
      elasticsearch {
        index => "%{[data_stream][type]}-custom-%{[data_stream][namespace]}"
        action => "create"
        cloud_id => "${CLOUD_ID}"
        cloud_auth => "${CLOUD_AUTH}"
      }
    } else {
      elasticsearch {
        data_stream => "true"
        cloud_id => "${CLOUD_ID}"
        cloud_auth => "${CLOUD_AUTH}"
      }
    }
  }
}
```

<img width="1254" alt="Screenshot 2023-07-13 at 3 04 24 PM" src="https://github.com/elastic/apm-server/assets/5618806/5f12e603-e77a-484b-a0b7-6e21059ac17d">